examples/persons/src/FilterByNameCache.re

```diff
@@ -0,0 +1,91 @@
+module PersonsNameFilterConfig = [%graphql
+  {|
+  query getPersonsWithName($name: String!) {
+    allPersons(filter: { name: $name } ) {
+      id
+      name
+      age
+    }
+  }
+|}
+];
+
+external cast: Js.Json.t => PersonsNameFilterConfig.t = "%identity";
```

Comment:
thanks for this advanced demo!
Could you add some more comments about what is going on and why the casting and raw are necessary for this very cases? Probably some beginner will read this in the future as reference

Reply:
Yes, absolutely! I will prepare a PR with some comments on the source code.